pkg/controllers/namespace/subnet_share_controller.go

```diff
@@ -0,0 +1,253 @@
+package namespace
+
+import (
+	"context"
+	"fmt"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/vmware-tanzu/nsx-operator/pkg/apis/vpc/v1alpha1"
+	"github.com/vmware-tanzu/nsx-operator/pkg/controllers/common"
+)
+
+// createSubnetCRInK8s creates the Subnet CR in Kubernetes
+func (r *NamespaceReconciler) createSubnetCRInK8s(ctx context.Context, subnetCR *v1alpha1.Subnet, subnetName string) error {
+	err := r.Client.Create(ctx, subnetCR)
+	if err != nil {
+		// If the Subnet CR already exists with the same name, try using generateName
+		if apierrors.IsAlreadyExists(err) {
+			log.Info("Subnet CR with the same name already exists, using generateName",
+				"Namespace", subnetCR.Namespace, "Name", subnetName)
+
+			// Create a new Subnet CR with generateName
+			// subnetCR.ObjectMeta.Name will be subnetName + "-" + randomSuffix
+			subnetCR.ObjectMeta.Name = ""
+			subnetCR.ObjectMeta.GenerateName = subnetName + "-"
+
+			err = r.Client.Create(ctx, subnetCR)
+			if err != nil {
+				return fmt.Errorf("failed to create Subnet CR with generateName: %w", err)
+			}
+		} else {
+			log.Error(err, "Failed to create Subnet CR", "Namespace", subnetCR.Namespace, "Name", subnetName)
+			return fmt.Errorf("failed to create Subnet CR: %w", err)
+		}
+	}
+
+	return nil
+}
+
+// createSharedSubnetCR creates a Subnet CR for a shared subnet
+func (r *NamespaceReconciler) createSharedSubnetCR(ctx context.Context, ns string, sharedSubnetPath string) error {
+	// Extract the org id, project id, VPC id, and subnet id
+	orgID, projectID, vpcID, subnetID, err := common.ExtractSubnetPath(sharedSubnetPath)
+	if err != nil {
+		return err
+	}
+
+	projectName, err := r.VPCService.GetProjectName(orgID, projectID)
+	if err != nil {
+		return fmt.Errorf("failed to get project name: %w", err)
+	}
+
+	vpcName, err := r.VPCService.GetVPCName(orgID, projectID, vpcID)
+	if err != nil {
+		return fmt.Errorf("failed to get VPC name: %w", err)
+	}
+
+	// Format VPC full name
+	vpcFullName := fmt.Sprintf("%s:%s", projectName, vpcName)
+	if isDefault, err := r.SubnetService.IsDefaultNSXProject(orgID, projectID); err != nil {
+		return fmt.Errorf("failed to check if project is default: %w", err)
+	} else if isDefault {
+		vpcFullName = fmt.Sprintf(":%s", vpcName)
+	}
```

Comment:
can we make it a func to get the vpcName? since the same logic in subnet_controller.go

Reply:
added